### DISCUSSION TITLE
Use idrac-virtualmedia driver to provision a host

### DISCUSSION THREAD

**Discussion by xisheng** [metal3-io/metal3-dev-env#1386] [Q&A] [answered]
Hello, team
I am trying the metal3 with driver idrac-virtualmedia to provision a dell physical host. the ISO image has been put onto a http server which could be downloaded successfully (like http://10.1.1.2/image/test.iso), however when I tried to provision the host with baremetalhost specs, ironic always tried to use the iso image from http://172.22.0.2:6180, instead of the image from the specified http server even I specified the url/checksum/checksumType in image section of baremetalhost spec. is there any other configuration I should do?

**Comment by dtantsur** [comment, marked as answer]
> ironic always tried to use the iso image from http://172.22.0.2:6180/

Ironic always caches images to serve them from the provisioining network. This is by design - you should not expect your BMCs to access arbitrary networks. Please make sure this IP and port are reachable.

**Reply by xisheng** [reply]
Thanks a lot @dtantsur for your quick response. is there a way to config the ironic to create the image based on specified iso, or cache the specified iso? Per my trying, the iso I specified is not used, instead it created the iso based on "kernel http://172.22.0.2:6180/images/ironic-python-agent.kernel, ramdisk http://172.22.0.2:6180/images/ironic-python-agent.initramfs, bootloader http://172.22.0.2:6180/uefi_esp.img".
Please take below for your reference
Details with "kubectl get baremetalhost":
spec:
  automatedCleaningMode: metadata
  bmc:
    address: idrac-virtualmedia://10.51.173.105/redfish/v1/Systems/System.Embedded.1
    credentialsName: idrac-credentials
    disableCertificateVerification: true
  bootMACAddress: xx:xx:xx:xx:xx:xx
  bootMode: UEFI
  image:
    checksum: 09C10B701DE39854BB799BCB975DDEFD
    checksumType: md5
    url: http://10.1.1.2/iso/custom.iso
  online: true
status:
  errorCount: 1
  errorMessage: 'Failed to inspect hardware. Reason: unable to start inspection: (''All
    virtual media mount attempts failed. Most recent error: '', (''Inserting virtual
    media into %(boot_device)s failed for node %(node)s, moving to next virtual media
    device, if available'', {''node'': ''aa3c04a6-0b4c-4315-b5d9-fcf5059afa5e'', ''boot_device'':
    <VirtualMediaType.CD: ''CD''>}))'
  errorType: inspection error
Checking ironic log as below:
2024-04-23 05:09:25.943 1 DEBUG ironic.drivers.modules.image_utils [None req-c232b773-63ec-460f-b64d-c8940e3b00a0 - - - - - -] Trying to create uefi ISO image for node aa3c04a6-0b4c-4315-b5d9-fcf5059afa5e with kernel http://172.22.0.2:6180/images/ironic-python-agent.kernel, ramdisk http://172.22.0.2:6180/images/ironic-python-agent.initramfs, bootloader http://172.22.0.2:6180/uefi_esp.img and kernel params nofb nomodeset ......
Any parameter we need to config in spec to leverage specified iso? actually I just want to use it directly, do NOT need ironic re-create it again.

also another questions from me is since the 172.22.0.2 is the ironic pod address which cannot be accessed from external host(10.1.1.x) to mount in virtual media. how to to config the address to have it accessed by external physical host? Thanks a lot
